### PR TITLE
Adding new mesh modifier to select nodesets using bounding box

### PR DESCRIPTION
**Pull request by amburan**
Description: Adding new mesh modifier to select nodesets using a bounding box. 
Issue number: #5340

### CODE REVIEW

**Comment by jwpeterson** [comment]
This should Ref #5340

File: framework/src/meshmodifiers/BoundingBoxNodeSet.C [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  InputParameters params = validParams<MeshModifier>();
+  params.addRequiredParam<std::vector<BoundaryName> >("new_boundary", "The name of the nodeset to create");
+  params.addParam<RealVectorValue>("bottom_left", "The bottom left point (in x,y,z with spaces in-between) of the box which contains the centroids of the elements whose nodes will be selected.");
+  params.addParam<RealVectorValue>("top_right", "The bottom left point (in x,y,z with spaces in-between) of the box which contains the centroids of the elements whose nodes will be selected.");
+  params.addParam<MooseEnum>("location", location, "Control of where the nodeset is to be set");
+
+  return params;
+}
+
+///
```
Comment (aeslaughter) [inline framework/src/meshmodifiers/BoundingBoxNodeSet.C]: This empty comment should be removed

File: framework/src/meshmodifiers/BoundingBoxNodeSet.C [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    bool found_node = false;
+    // Loop over the elements and assign node set id to nodes within the bounding box
+    for (MeshBase::element_iterator el = mesh.active_elements_begin(); el != mesh.active_elements_end(); ++el)
+    {
+      const Elem* elem = *el;
+
+      bool contains = _bounding_box.contains_point((*el)->centroid());
+      if ((contains && _location == "INSIDE") || (!contains && _location == "OUTSIDE"))
+      {
+        //(*el)->subdomain_id() = _block_id;
```
Comment (aeslaughter) [inline framework/src/meshmodifiers/BoundingBoxNodeSet.C]: Please remove this commented code

File: framework/src/meshmodifiers/BoundingBoxNodeSet.C [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+      }
+    }
+    if (!found_node)
+      mooseError("No element centroids found within the bounding box!");
+  }
+  else
+  {
+    mooseError("Node set can not be empty!");
+  }
```
Comment (aeslaughter) [inline framework/src/meshmodifiers/BoundingBoxNodeSet.C]: We typically do not use braces for oneliners

**Comment by aeslaughter** [comment]
This looks great, I made a few minor recommendations. If you fix those up I will get this merged.

Thanks!

**Comment by amburan** [comment]
John, can you tell me how to change the reference to #5344 ?

**Comment by permcody** [comment]
This looks good, you referenced the ticket by putting it in your commit message preceded by a pound sign. For future reference you can use the keyword "closes" before the pound sign and ticket to actually close the ticket automatically once your PR is merged. Don't worry about updating this PR though.

Thanks for the contribution!

**Pull request merged**